multicamera_acquisition/visualization.py

```diff
@@ -214,4 +263,72 @@ def plot_video_stats(csv_path, name):
     print(f"Total time elapsed: {time_elapsed} seconds")
     print(f"Average framerate: {1 / (avg_diffs* 1e-9)} Hz")
 
-    return
\ No newline at end of file
+    return
+
+
+def plot_image_grid(images, display_config):
+    """
+    Parameters
+    ----------
+    images : dict
+        Mapping from camera names to images from that camera
+    display_config : dict
+        Config dictionary for a MultiDisplay
+    """
+
+    cfg = display_config
+    nrow = int(np.ceil(len(cfg['camera_names']) / cfg['cameras_per_row']))
+    fig, ax = plt.subplots(nrow, cfg['cameras_per_row'],
+        figsize = (2 * cfg['cameras_per_row'], 2 * nrow))
+    ax = ax.ravel()
+
+    # plot image for each camera in display config, formatted as in Multidisplay
+    for a, camera_name, rng in zip(ax, cfg['camera_names'], cfg['ranges']):
+        frame = images[camera_name]
+        frame = format_frame(
+            frame,
+            cfg['downsample'],
+            cfg['size'],
+            rng,
+            frame.dtype == np.uint16 or ("lucid" in camera_name))
+        a.imshow(frame)
+        a.set_title(camera_name)
+        a.set_xticks([])
+        a.set_yticks([])
+    
+    # hide unused axes
+    for a in ax[len(cfg['camera_names']):]:
+        a.set_axis_off()
+
+    fig.tight_layout()
+    return fig, ax
+
+
+def load_first_frames(location):
+    """
+    Load first frame of an acquisition for visualization
+
+    Parameters
+    ----------
+    location : Path
+        Directory passed to `acquire_video`
+    Returns
+    -------
+    images : dict[str, array]
+        Mapping of camera name to first frame of acquired video
+    """
+    
+    images = {}
+    files = list(glob.glob(str(location / "*.mp4")))
+    if len(files) == 0:
+        logging.log(logging.WARN, f"No recordings found at {location}")
+    for f in files:
+        basename = os.path.basename(f)
+        cam_name = basename.split('.')[-2]
+        cap = cv2.VideoCapture(f)
```

Comment:
So is the idea that to check all is good with the cameras, you would first capture a very short vid, then use this and plot_image_grid to display the first frames? If so, one possible issue is that it will (probably) fail for un-muxed NVC videos. Worth a try.

Reply:
Good point, as we ok to require that automux was on for the short capture?

Reply:
yeah, or we could use ffmpeg